### PR TITLE
Document and verify that claims in oe_verify_evidence() is optional

### LINKED ISSUE #3594: Confirm and check that the parameter `claims` in API `oe_verify_evidence()` is optional
In `include/openenclave/attestation/verifier.h`, the `claims` parameter in API `oe_verify_evidence()` is not documented as optional, but the implementation of this API in `common/attest_plugin.c` allows this parameter to be NULL. On the other hand, `_verify_evidence()` in `common/sgx/verifier.c` requires its `claims` parameter to be non-NULL.

Need to clarify doxygen documentation and update implementation to support the `claims` parameter in API `oe_verify_evidence()`as optional, and to skip calling `oe_sgx_extract_claims()` when this parameter is NULL.

This issue was brought up in this [comment thread](https://github.com/openenclave/openenclave/pull/3499#discussion_r494672966).

### PR DESCRIPTION
**Pull request by shnwc**
Fix #3594

### CODE REVIEW

**Comment by shnwc** [comment]
bors try

File: common/sgx/verifier.c [hunk 1]
```diff
@@ -456,8 +456,9 @@ oe_result_t oe_sgx_extract_claims(
 
     // Note: some callers can have custom_claims_buffer pointing to a non-NULL
     // buffer containing a zero-sized array.
-    if (!format_id || !report_body || !report_body_size ||
-        (!custom_claims_buffer && custom_claims_buffer_size))
+    if (!format_id || !report_body || !report_body_size || !claims_out ||
+        !claims_length_out ||
+        (!custom_claims_buffer != !custom_claims_buffer_size))
```
Comment (dthaler) [inline common/sgx/verifier.c:461]: This PR doesn't touch include\openenclave\internal\sgx\plugin.h but that's where the doxygen for oe_sgx_extract_claims() is and `custom_claims_buffer` and size are not stated as optional there, so I think we need to update that file too (I will also observe that the `sgx_endorsements` param is out of order in the param list in the doxygen so if you update the file maybe fix that too)

Comment (shnwc) [reply common/sgx/verifier.c:461]: Good catch! I'll fix. On the other hand, `include\openenclave\internal\sgx\plugin.h` is internal, so OE SDK application developers don't see this header file or its doxygen document.

**Comment by shnwc** [comment]
bors try

File: include/openenclave/internal/sgx/plugin.h [hunk 2]
```diff
@@ -33,11 +33,11 @@ typedef enum _sgx_evidence_format_type_t
...
  * @param[in] report_body Pointer to a buffer with raw SGX quote / report.
  * @param[in] report_body_size Size of the report_body buffer.
- * @param[in] sgx_endorsements Pointer to the endorsements buffer.
- * @param[in] custom_claims_buffer The data in a flat buffer to be packaged as a
- * claim of ID OE_CLAIM_CUSTOM_CLAIMS_BUFFER when format_type has the right
- * value.
+ * @param[in] custom_claims_buffer If not NULL, it holds the data in a flat
+ * buffer to be packaged as a claim of ID OE_CLAIM_CUSTOM_CLAIMS_BUFFER when
+ * format_type has the right value.
  * @param[in] custom_claims_buffer_size The size of the custom_claims buffer.
```
Comment (dthaler) [inline include/openenclave/internal/sgx/plugin.h:39]: Shouldn't the size param also say "If not NULL, ..."?

Comment (shnwc) [reply include/openenclave/internal/sgx/plugin.h:39]: `custom_claims_buffer_size` is of type `size_t`, so it can't be NULL.

**Review by dthaler — APPROVED** [review]

**Review by yentsanglee — APPROVED** [review]

**Review by qiucwang — APPROVED** [review]

**Pull request merged**